docs/recommendations.md

```diff
@@ -1,18 +1,139 @@
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Prepare container metadata
+        id: meta
+        uses: docker/metadata-action@8e5442c4ef9f78752691e2d8f8d19755c6f78e81 # v5.5.1
+        with:
+          images: |
+            ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          tags: |
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
+            type=semver,pattern={{major}}
+            type=sha
+
+      - name: Build and push the images
+        id: push
+        uses: docker/build-push-action@a254f8ca60a858f3136a2f1f23a60969f2c402dd # v6.4.0
+        with:
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+
+      - name: Attest container images
+        uses: actions/attest-build-provenance@5e9cb68e95676991667494a6a4e59b8a2f13e1d0 # v1.3.3
+        id: attest
+        with:
+          subject-name: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          subject-digest: ${{ steps.push.outputs.digest }}
+          push-to-registry: true
+```
+
+Your users would then be able to verify it using `gh`:
+
+```command
+gh attestation verify my-artifact.tar.gz -o open-telemetry
+```
+
+The container image can be verified with:
+
+```command
+gh attestation verify oci://ghcr.io/open-telemetry/my-sig-deliverable:0.0.1 -o open-telemetry
+```
+
+### How it works
+
+We recommend the following resources to understand in detail how it works:
+
+- GitHub's blog post [Introducing Artifact Attestations–now in public
+  beta](https://github.blog/2024-05-02-introducing-artifact-attestations-now-in-public-beta/).
+- [`sigstore` project documentation](https://docs.sigstore.dev/), altough it makes it sound more complicated than it
+  really is.
+
+In particular, this is how it works on our setup:
+
+- When your software is being released as part of a GitHub Workflow, the GitHub Action (or `goreleaser`, see [Go](#go))
+  obtains an OIDC token from GitHub, which contains the organization, repository, and build information (workflow
+  execution, git reference, ...)
+- Your artifact(s) is signed with an ephemeral key, with the token metadata being part of it (issuer, identity, ...)
+- The results of this action are published on a transparency log, like this for the [container image for OTel Collector
+  Contrib v0.104.0](https://search.sigstore.dev/?logIndex=107849861)
+
+### Go
+
+For Go projects, we have success using the signing integration in `goreleaser`. This uses `cosign` (from `sigstore`) as
+well, consistent with our recommendation. More recently, GitHub published an improved support for provenance
+attestations, which are the base for our recommendation.
+
+### Other languages
+
+We do not currently have other usage of `sigstore` as part of the project. The links below are not official
```

Comment:
@ThomsonTan would you share more about what you've tried in OpenTelemetry .NET + sigstore?

Reply:
sgtm